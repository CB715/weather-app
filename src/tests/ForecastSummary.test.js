import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 12345,
    description: "Stub description",
    icon: "Stub icon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
