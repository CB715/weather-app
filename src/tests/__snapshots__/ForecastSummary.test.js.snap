// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    >
      12345
    </div>
    <div
      class="forecast-summary_temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary_description"
    >
      Stub description
    </div>
    <div
      class="forecast-summary_icon"
    >
      Stub icon
    </div>
  </div>
</DocumentFragment>
`;
